src/App.jsx

```diff
@@ -14,45 +17,109 @@ export const goodsFromServer = [
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_GOODS_ALPH = 'alphabetically';
+const SORT_GOODS_LENGTH = 'length';
+const GOODS_REVERSE = 'reverse';
+
+function getPreparedGoods(goods, { sortFilter, reversedGoods }) {
+  const preperedGoods = [...goods];
+
+  if (sortFilter) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortFilter) {
+        case SORT_GOODS_ALPH:
+          return good1.localeCompare(good2);
+
+        case SORT_GOODS_LENGTH:
+          return good1[sortFilter] - good2[sortFilter];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversedGoods) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortFilter, setSortFilter] = useState('');
+  const [reversedGoods, setReversedGoods] = useState('');
+
+  const vissibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortFilter, reversedGoods },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortFilter !== SORT_GOODS_ALPH },
+          )}
+          onClick={() => setSortFilter(SORT_GOODS_ALPH)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortFilter !== SORT_GOODS_LENGTH },
+          )}
+          onClick={() => setSortFilter(SORT_GOODS_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': reversedGoods !== GOODS_REVERSE },
+          )}
+          onClick={() => (
+            reversedGoods === '' ? (
+              setReversedGoods(GOODS_REVERSE)
+            ) : (
+              setReversedGoods('')
+            )
+          )}
+        >
+          Reverse
+        </button>
+
+        { (sortFilter !== '' || reversedGoods !== '') && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortFilter('');
+              setReversedGoods('');
+            }}
```

Comment:
Please move to separate function instead of useing anonymous function. This way your code will be more maintainable, and easier to extend and reuse.